x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/all/index.tsx

```diff
@@ -36,43 +37,47 @@ import { PrePackagedRulesPrompt } from '../components/pre_packaged_rules/load_em
 import { RuleDownloader } from '../components/rule_downloader';
 import { getPrePackagedRuleStatus } from '../helpers';
 import * as i18n from '../translations';
-import { EuiBasicTableOnChange, TableData } from '../types';
+import { EuiBasicTableOnChange } from '../types';
 import { getBatchItems } from './batch_actions';
 import { getColumns } from './columns';
 import { showRulesTable } from './helpers';
 import { allRulesReducer, State } from './reducer';
 import { RulesTableFilters } from './rules_table_filters/rules_table_filters';
 
+// EuiBasicTable give me a hardtime with adding the ref attributes so I went the easy way
+// after few hours of fight with typescript !!!! I lost :(
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const MyEuiBasicTable = styled(EuiBasicTable as any)`` as any;
```

Comment:
Fiddled with this one as well and wasn't able to get anywhere. 😔 Would be nice if we could `ts-ignore` just the `ref` prop and keep the type-checking for the remainder of the component as mentioned [here](https://github.com/microsoft/TypeScript/issues/33795), but that's not an option it looks like. Necessary evil at this point to ensure we have the ability to clear selection and fix additional bugs. Thanks for the comment for our future selves 👍